js/addRow.js

```diff
@@ -0,0 +1,34 @@
+const totalColumnToAdd = () => {
+  const tableRow = document.getElementsByTagName("tr")[0]
+  if(tableRow){
+    const childrenCount = tableRow.children.length
+    return childrenCount === 0 ? initial.gridDims.cols : childrenCount
+  }
+  return initial.gridDims.cols
+}
+
+const getTableBody = () => {
+  const tbodyEntries = document.getElementsByTagName('tbody')
+  if(tbodyEntries.length === 0) return
+  return tbodyEntries[0]
+}
+
+const buildRow = (cellCount) => {
+  const tableRow = document.createElement("tr")
+  for(let cell = 0 ; cell < cellCount ; cell++){
+    // This could could be reusable but oh well
```

Comment:
We do have a `populateGrid()` in the respectively named file. Could that help?

Reply:
Yah, I was referring to how rows and cells were created there. Since the cells all share the same class name and style, it's a bit of repeating code. We could've extracted that to it's own function. But I got lazy